lib/vector-common/src/internal_event/component_events_dropped.rs

```diff
@@ -0,0 +1,44 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[allow(dead_code)]
+pub const INTENTIONAL: bool = true;
+pub const UNINTENTIONAL: bool = false;
+
+#[derive(Debug)]
+pub struct ComponentEventsDropped<const INTENTIONAL: bool> {
+    pub count: u64,
+    pub reason: &'static str,
+}
+
+impl<const INTENTIONAL: bool> InternalEvent for ComponentEventsDropped<INTENTIONAL> {
+    fn emit(self) {
+        let message = "Events dropped";
+        if INTENTIONAL {
+            debug!(
+                message,
+                intentional = INTENTIONAL,
+                reason = self.reason,
+                count = self.count,
+                internal_log_rate_secs = true,
+            );
+        } else {
+            error!(
+                message,
+                intentional = INTENTIONAL,
```

Comment:
```suggestion
                intentional = UNINTENTIONAL,
```

Unless I'm missing something? Both the if and else lead to intentional.

Reply:
Good catch- I moved this struct over from `src/internal_events/common.rs` , and that bug was not caught during the original PR :) I'll fix it.

Reply:
AN UNINTENTIONAL MISTAKE 👀

Reply:
Just realized actually that this <WAS INTENTIONAL> :roll_eyes: 

We are overridding the const INTENTIONAL from the file in this function with the one set in the function definition.

I'll change the name so it is more clear.